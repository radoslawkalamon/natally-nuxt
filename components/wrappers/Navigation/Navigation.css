.navigation {
  display: flex;
  flex-flow: column nowrap;
  gap: var(--gap);
  list-style: none;
  margin: 0;
  padding: 0;
}
