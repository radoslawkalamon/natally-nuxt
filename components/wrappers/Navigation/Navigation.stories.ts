import type { Meta, Story } from '@storybook/vue'
import WrappersNavigation from '@/components/wrappers/Navigation/Navigation.vue'

const meta: Meta = {
  title: 'Wrappers / Navigation',
  component: WrappersNavigation,
  argTypes: {
    default: {
      description: 'Should be used with `<ButtonText>` and `<LinkText>`',
      table: {
        type: null,
        defaultValue: null
      }
    }
  }
}
export default meta

export const Default: Story = () => ({
  components: { WrappersNavigation },
  template: `<WrappersNavigation>
    <span>Child 1</span>
    <span>Child 2</span>
  </WrappersNavigation>`
})
