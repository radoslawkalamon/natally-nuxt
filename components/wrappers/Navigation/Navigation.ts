import Vue from 'vue'

export default Vue.extend({
  name: 'WrappersNavigation',
  components: {
    VNode: {
      functional: true,
      // @ts-ignore: This is hacky as hell but it works :D
      render: (_h, { props }) => props.node
    }
  },
  computed: {
    children () {
      return (this.$slots.default || [])
        .filter(child => !child.text)
    }
  }
})
