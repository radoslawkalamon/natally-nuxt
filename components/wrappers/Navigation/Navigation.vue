<template>
  <ul
    v-if="children.length !== 0"
    class="navigation"
  >
    <template v-for="(child, index) in children">
      <li
        :key="index"
        class="navigation__item"
      >
        <vnode :node="child" />
      </li>
    </template>
  </ul>
</template>

<script src="./Navigation.ts" lang="ts"></script>
<style src="./Navigation.css" scoped></style>
