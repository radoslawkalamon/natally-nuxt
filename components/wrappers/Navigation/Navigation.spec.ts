import Navigation from '@/components/wrappers/Navigation/Navigation.vue'
import { shallPassIntegrationSanityTest } from '@/devtools/jest.common.spec'

describe('Wrappers / Navigation', () => {
  describe('Integration', () => {
    shallPassIntegrationSanityTest({
      component: Navigation,
      description: 'with children',
      options: {
        slots: {
          default: [
            { template: '<span>Tag child 1</span>' },
            'Text child',
            { template: '<span>Tag child 2</span>' }
          ]
        }
      }
    })

    shallPassIntegrationSanityTest({
      component: Navigation,
      description: 'no children'
    })
  })
})
