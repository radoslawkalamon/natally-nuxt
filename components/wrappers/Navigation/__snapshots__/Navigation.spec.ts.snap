// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrappers / Navigation Integration shall pass integration sanity test | no children 1`] = `""`;

exports[`Wrappers / Navigation Integration shall pass integration sanity test | with children 1`] = `
"<ul class=\\"navigation\\">
  <li class=\\"navigation__item\\"><span>Tag child 1</span></li>
  <li class=\\"navigation__item\\"><span>Tag child 2</span></li>
</ul>"
`;
